File: src/main/java/com/lannstark/lec15/Lec15Main.java
package com.lannstark.lec15;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Map.Entry;

public class Lec15Main {

    public static void main(String[] args) {
        int[] array = {100, 200};

        for (int i = 0; i < array.length; i++) {
            System.out.printf("%s, %s", i, array[i]);
        }

        //JDK 8
        final List<Integer> numbers = Arrays.asList(100, 200);

        //하나 가져오기
        System.out.println(numbers.get(0));

        //For Each
        for (Integer number : numbers) {
            System.out.println(number);
        }

        //전통적인 For문
        for (int i = 0; i < numbers.size(); i++) {
            System.out.printf("%s, %s", i, numbers.get(i));
        }

        // JDK 8
        Map<Integer, String> oldMap = new HashMap<>();
        oldMap.put(1, "MONDAY");
        oldMap.put(2, "TUESDAY");

        // JDK 9
        Map.of(1, "MONDAY", 2, "TUESDAY");

        for (Integer key : oldMap.keySet()) {
            System.out.println(key);
            System.out.println(oldMap.get(key));
        }

        for (Entry<Integer, String> entry : oldMap.entrySet()) {
            System.out.println(entry.getKey());
            System.out.println(entry.getValue());
        }
    }

}

File: src/main/java/com/lannstark/lec15/Lec15Main.kt
package com.lannstark.lec15

fun main() {
    val array = arrayOf(100, 200)
    array.plus(300) // java에서는 copy해서..

    for (i in array.indices) {
        println("${i}, ${array[i]}")
    }

    for ((idx, value) in array.withIndex()) {
        println("${idx}, ${value}")
    }

    // 불변컬렉션, 가변컬렉션
    val numbers1 = listOf(100, 200) // 불변리스트
    val numbers2 = mutableListOf(100, 200) // 가변리스트

    val emptyList = emptyList<Int>() // 타입 초기화

    //하나 가져오기
    println(numbers1.get(0))
    println(numbers1[0])

    //For Each
    for (number in numbers1) {
        println(number)
    }

    //전통적인 For문
    for ((idx, value) in numbers1.withIndex()) {
        println("${idx}, ${value}")
    }

    val set1 = setOf(1,2)
    val set2 = mutableSetOf(1,2)

    val oldMap = mutableMapOf<Int, String>()
    //put 사용 가능
    oldMap[1] = "MONDAY"
    oldMap[2] = "TUESDAY"

    mapOf(1 to "MONDAY", 2 to "TUESDAY")

    for (key in oldMap.keys) {
        println(key)
        println(oldMap[key])
    }

    for ((key, value) in oldMap.entries) {
        println(key)
        println(value)
    }

    // List<Int?> : 리스트에 null 들어갈 수 있지만, 리스트는 절대 null 아님
    // List<Int>? : 리스트에 null이 들어갈 수 없지만 리스트가 null일 수 있음
    // List<Int?>? : 리스트에 null이 들어갈 수 있고, 리스트가 null 일 수 있음







}